### PR TITLE
fix(docs): Correct broken links to Red Hat documentation

### PR DESCRIPTION
**Pull request by butler54**
**- Description of the problem which is fixed/What is the use case**
There are broken links which are visible in the operator catalog when installing the Sandbox container operator on OCP 4.16

**- What I did**
Found an appropriate replacement link which is not version specific and should be stable moving forward

**- How to verify it**
Check the links before and after.

**- Description for the changelog**
Update broken documentation links for operator config and readme

### CODE REVIEW

**Review by spotlesstofu — APPROVED** [review]
Thanks!

File: config/manifests/bases/sandboxed-containers-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -106,7 +106,7 @@ spec:
         on "create instance" on the operator overview page.
 
     # Documentation
-    See the official documentation [here](https://docs.openshift.com/container-platform/latest/sandboxed_containers/understanding-sandboxed-containers.html)
+    See the official documentation [here](https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/)
```
Comment (apinnick) [inline config/manifests/bases/sandboxed-containers-operator.clusterserviceversion.yaml]:
```suggestion
    See the [OpenShift sandboxed containers documentation](https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/).
```

File: README.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ Please refer to the [code](https://github.com/openshift/sandboxed-containers-ope
 ### Getting Started
 
 Please refer to the OpenShift release specific documentation for getting started with sandboxed containers.
```
Comment (apinnick) [inline README.md:61]:
```suggestion
See the [OpenShift sandboxed containers documentation](https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/) to get started with sandboxed containers. 
```

File: README.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ Please refer to the [code](https://github.com/openshift/sandboxed-containers-ope
 ### Getting Started
 
 Please refer to the OpenShift release specific documentation for getting started with sandboxed containers. 
-- For OpenShift latest documentation please follow this [doc](https://docs.openshift.com/container-platform/latest/sandboxed_containers/deploying-sandboxed-container-workloads.html)
+- For OpenShift latest documentation please follow this [doc](https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/)
```
Comment (apinnick) [inline README.md]:
```suggestion
```

**Comment by butler54** [comment]
Updated with @apinnick's changes

**Review by gkurz — APPROVED** [review]
/lgtm

Thanks @butler54 and @apinnick !

I'll squash the PR into a single commit at merge time.

**Pull request merged**